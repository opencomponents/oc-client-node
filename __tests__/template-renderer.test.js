'use strict';

const TemplateRenderer = require('../src/template-renderer');

const templateRenderer = new TemplateRenderer(); // Q.: this is totally unecessary

test('TemplateRenderer ctor', () => {
  expect(typeof templateRenderer).toEqual('function');
});

const scenarios = [
  {
    description: 'bad template type',
    template: null,
    model: null,
    options: { templateType: 'unsupported' }
  },
  {
    description: 'jade',
    template: null,
    model: null,
    options: { templateType: 'jade' }
  },
  {
    description: 'handlebars',
    template: null,
    model: null,
    options: { templateType: 'handlebars' }
  }
];

scenarios.forEach(({ description, template, model, options }) => {
  test(description, () => {
    const callback = jest.fn();
    templateRenderer(template, model, options, callback);
    expect(callback).toHaveBeenCalled();
    expect(callback.mock.calls).toMatchSnapshot();
  });
});
