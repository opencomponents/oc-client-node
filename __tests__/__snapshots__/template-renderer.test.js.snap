// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bad template type 1`] = `
Array [
  Array [
    "Error requiring oc-template: \\"unsupported\\" not found",
  ],
]
`;

exports[`handlebars 1`] = `
Array [
  Array [
    [TypeError: Cannot read property 'compiler' of null],
  ],
]
`;

exports[`jade 1`] = `
Array [
  Array [
    [TypeError: options.template is not a function],
    "<oc-component href=\\"{0}\\" data-hash=\\"{1}\\" data-name=\\"\\" data-rendered=\\"true\\" data-version=\\"{3}\\">{4}</oc-component>",
  ],
]
`;
